extensions/binding/org.eclipse.smarthome.binding.lifx/src/main/java/org/eclipse/smarthome/binding/lifx/internal/LifxNetworkThrottler.java

```diff
@@ -25,8 +27,8 @@
 
     public final static long PACKET_INTERVAL = 50;
 
-    private static ConcurrentHashMap<String, ReentrantLock> locks = new ConcurrentHashMap<String, ReentrantLock>();
-    private static ConcurrentHashMap<String, Long> timestamps = new ConcurrentHashMap<String, Long>();
+    private static Map<String, ReentrantLock> locks = Collections.synchronizedMap(new HashMap<String, ReentrantLock>());
+    private static Map<String, Long> timestamps = Collections.synchronizedMap(new HashMap<String, Long>());
 
     public static void lock(String key) {
         if (!locks.containsKey(key)) {
```

Comment:
I don't think it is thread safe to first inspect the map and then add the key in the next statement. Most of the time it will work though. ;-) But in theory two threads may inspect the map. Conclude that the key is not there, and then one thread would overwrite the others value.